src/main/webapp/app/entities/flair-bi/flair-bi-content-header.controller.js

```diff
@@ -652,22 +652,24 @@
 
         function removeFilterTag($event, val, list, key) {
             $event.preventDefault();
-            var index = list.indexOf(val);
-            if (index > -1) {
-                list.splice(index, 1);
-            }
-            vm.filters[key] = list;
-            removeTagInBI({ 'text': val });
-            setNoOfPages();
             Views.getCurrentEditState({
                 id: $stateParams.id
             },
                 function (result, headers) {
-                    vm.iFrames = result.visualMetadataSet.filter(function (item) {
-                        return item.metadataVisual.name === IFRAME.iframe;
-                    })
+                    $rootScope.updateWidget = {};
+                    //comment for now 
+                    // vm.iFrames = result.visualMetadataSet.filter(function (item) {
+                    //     return item.metadataVisual.name === IFRAME.iframe;
+                    // })
+                    var index = list.indexOf(val);
+                    if (index > -1) {
+                        list.splice(index, 1);
+                    }
+                    vm.filters[key] = list;
                     var filters = filterParametersService.get();
-                    filterParametersService.setFilterInIframeURL(filters, vm.iframes, vm.dimension);
+                    filterParametersService.setFilterInIframeURL(filters, vm.iFrames, vm.dimensions);
```

Comment:
This change has been suggested 2 to 3 times over github and Skype chat. Filters object already exist in service so no need to pass